src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -212,29 +190,109 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
             }
 
             logger.Info(ErrorCode.SiloInitializing, "-------------- Initializing {0} silo on host {1} MachineName {2} at {3}, gen {4} --------------",
-                siloType, nodeConfig.DNSHostName, Environment.MachineName, here, generation);
+                this.initializationParams.Type, LocalConfig.DNSHostName, Environment.MachineName, localEndpoint, this.initializationParams.SiloAddress.Generation);
             logger.Info(ErrorCode.SiloInitConfig, "Starting silo {0} with the following configuration= " + Environment.NewLine + "{1}",
                 name, config.ToString(name));
-
-            if (keyStore != null)
-            {
-                // Re-establish reference to shared local key store in this app domain
-                LocalDataStoreInstance.LocalDataStore = keyStore;
-            }
-
+            
             // Configure DI using Startup type
-            bool usingCustomServiceProvider;
-            Services = StartupBuilder.ConfigureStartup(nodeConfig.StartupTypeName, out usingCustomServiceProvider);
+            this.Services = StartupBuilder.ConfigureStartup(
+                this.LocalConfig.StartupTypeName,
+                (services, usingCustomServiceProvider) =>
```

Comment:
I have concerns about adding all of these classes to DI as singletons.
Singletons should be thread safe, as we've not control over who users them.  I do not believer many of these classes are thread safe.
Many of these singletons have very specific initialization logic that must be run before they are usable, and little to no testing done on what happens if they are used before they are initialized.
It is not clear to me which of these should remain singletons.

I don't think these concerns should block the change, I just want us to be aware of the risk we are introducing.